src/core/Subscription.ts

```diff
@@ -0,0 +1,126 @@
+import {
+	AudioPlayer,
+	AudioPlayerStatus,
+	AudioResource,
+	createAudioPlayer,
+	entersState,
+	VoiceConnection,
+	VoiceConnectionDisconnectReason,
+	VoiceConnectionStatus,
+} from '@discordjs/voice';
+
+import { Track } from './track';
+
+function wait(time: number) {
+	return new Promise((resolve) => setTimeout(resolve, time).unref());
+}
```

Comment:
Can be replaced with [`timersPromises.setTimeout(delay)`](https://nodejs.org/dist/latest-v16.x/docs/api/timers.html#timers_timerspromises_settimeout_delay_value_options) on Node.js 15+.